src/mongocrypt-crypto.c

```diff
@@ -352,114 +384,104 @@ _mongocrypt_memequal (const void *const b1, const void *const b2, size_t len)
    return ret;
 }
 
+typedef enum {
+   MODE_CBC,
+   MODE_CTR,
+} _mongocrypt_encryption_mode_t;
+
+typedef enum {
+   HMAC_NONE,
+   HMAC_SHA_512_256, // sha512 truncated to 256 bits
+   HMAC_SHA_256,
+} _mongocrypt_hmac_type_t;
+
+typedef enum {
+   KEY_FORMAT_FLE1, // 32 octets MAC key, 32 octets DATA key, 32 octets ignored
+   KEY_FORMAT_FLE2IEV, // 32 octets DATA key
+   KEY_FORMAT_FLE2, // 32 octets DATA key, 32 octets MAC key, 32 octets ignored
```

Comment:
I would call this `KEY_FORMAT_FLE2_AEAD` and the other one just `KEY_FORMAT_FLE2`. We using the AEAD variation in two cases (user values for unindexed and indexed) but the other one is used to protect all the tokens (ESC, ECOC)

Reply:
Done. Keeping consistent names is a good thing.